### PR TITLE
[Behat][Promotion] Percentage discount promotions scenarios

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets |
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: features/shop/apply_percentage_discount_promotion_on_order.feature [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        And the store has a product "PHP Mug" priced at "€6.00"
+        And the store has a product "Symfony Mug" priced at "€10.00"
+        And there is a promotion "Holiday promotion"
+        And it gives "20%" percentage discount for customers with carts above "€80.00"
+        And there is a promotion "Christmas sale"
+        And it gives "50%" percentage discount for customers with carts with more than 1 item
+
+    @todo
+    Scenario: Applying percentage discount based on cart total
+        And I have product "PHP T-Shirt" in the cart
```
Comment (michalmarcinkowski) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: Given

File: features/shop/apply_percentage_discount_promotion_on_order.feature [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        Then my cart total should be "€80.00"
+        And my cart promotions should be "-€20.00"
+
+    @todo
+    Scenario: Applying percentage discount based on items count
+        And I have product "PHP Mug" in the cart
```
Comment (michalmarcinkowski) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: Given

File: features/shop/apply_percentage_discount_promotion_on_order.feature [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        And I have product "Symfony Mug" in the cart
```
Comment (michalmarcinkowski) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: Given
When

File: features/shop/apply_percentage_discount_promotion_on_order.feature [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    Background:
+        Given the store is operating on a single channel
+        And default currency is "EUR"
+        And the store has a product "PHP T-Shirt" priced at "€100.00"
+        And there is a promotion "Holiday promotion"
+        And it gives "20%" percentage discount to every order
+
+    @todo
+    Scenario: Applying percentage discount to order
+        Given I have product "PHP T-Shirt" in the cart
```
Comment (michalmarcinkowski) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: When I add

File: features/shop/apply_percentage_discount_promotion_on_order.feature [hunk 5]
```diff
@@ -0,0 +1,18 @@
+@ui-cart
+Feature: Apply percentage discount promotion on order
+    In order to pay proper amount while buying promoted goods
+    As a Customer
+    I want to have promotions applied to my cart
+
+    Background:
+        Given the store is operating on a single channel
+        And default currency is "EUR"
```
Comment (michalmarcinkowski) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: Given the store operates on a single channel in "France"

**Comment by michalmarcinkowski** [comment]
I think we should add also the following scenario to this feature:
```
Scenario: The fixed discount does not affect a shipping fee
    Given the store has "DHL" shipping method with "€10.00" fee
    And I am logged in customer
    When I add product "PHP T-Shirt" to the cart
    And I proceed selecting "DHL" shipping method
    Then my cart total should be "€90.00"
    And my cart shipping fee should be "€10.00"
    And my cart promotions should be "-€20.00"
```

File: features/shop/apply_percentage_discount_promotion_on_order.feature [hunk 6]
```diff
@@ -0,0 +1,27 @@
+@ui-cart
```
Comment (lchrusciel) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: `@cart`

File: features/shop/apply_percentage_discount_promotion_on_order.feature [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        And it gives "20%" percentage discount to every order
+
+    @todo
+    Scenario: Applying percentage discount to order
+        When I add product "PHP T-Shirt" to the cart
+        Then my cart total should be "€80.00"
+        And my cart promotions should be "-€20.00"
+
+    @todo
+    Scenario: The percentage discount does not affect a shipping fee
```
Comment (pjedrzejewski) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: the shipping fee

Comment (pjedrzejewski) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: ``The`` before percentage discount is not needed as well.

File: features/shop/apply_percentage_discount_promotion_on_order.feature [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        And it gives "20%" percentage discount to every order
+
+    @todo
+    Scenario: Applying percentage discount to order
+        When I add product "PHP T-Shirt" to the cart
+        Then my cart total should be "€80.00"
+        And my cart promotions should be "-€20.00"
+
+    @todo
+    Scenario: percentage discount does not affect the shipping fee
```
Comment (peteward) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: As an admin user, if you wanted to discount shipping as well would you just add a secondary `action` (or whatever it's going to be called) to also discount the same percentage on shipping?

Comment (Zales0123) [inline features/shop/apply_percentage_discount_promotion_on_order.feature]: Yes, this is something you can do already, however, we want to ensure with this scenario about affecting nothing more than items total by *items* discount action ;)

**Comment by peteward** [comment]
:+1: just checking what was going to be possible

File: features/shop/promotion/apply_percentage_discount_promotion_on_order.feature [hunk 9]
```diff
@@ -0,0 +1,27 @@
+@promotion
+Feature: Apply percentage discount promotion on order
```
Comment (michalmarcinkowski) [inline features/shop/promotion/apply_percentage_discount_promotion_on_order.feature]: Receiving ...

File: features/shop/promotion/apply_percentage_discount_promotion_on_order.feature [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    In order to pay proper amount while buying promoted goods
+    As a Customer
+    I want to have promotions applied to my cart
```
Comment (michalmarcinkowski) [inline features/shop/promotion/apply_percentage_discount_promotion_on_order.feature]: I want to receive discount for my purchase

File: features/shop/promotion/receiving_percentage_discount_promotion_on_order.feature [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    @todo
+    Scenario: Receiving percentage discount for my cart
+        When I add product "PHP T-Shirt" to the cart
+        Then my cart total should be "€80.00"
+        And my cart promotions should be "-€20.00"
+
+    @todo
+    Scenario: Receiving percentage discount does not affect the shipping fee
+        Given the store has "DHL" shipping method with "€10.00" fee
+        And I am logged in customer
```
Comment (pamil) [inline features/shop/promotion/receiving_percentage_discount_promotion_on_order.feature]: Is it needed?

Comment (Zales0123) [inline features/shop/promotion/receiving_percentage_discount_promotion_on_order.feature]: To proceed order we should be logged in or proceed logging in or proceed as guest. However, I think that being logged in as a customer is the most basic case and it should be used here.

Comment (pjedrzejewski) [inline features/shop/promotion/receiving_percentage_discount_promotion_on_order.feature]: @Zales0123 :+1:

**Comment by pjedrzejewski** [comment]
:+1:

**Pull request merged**